### PR TITLE
[docs] req_subkeys - add documentation

### PR DESCRIPTION
**Pull request by mime-frame**
to: @jacknagz 
cc: @airbnb/streamalert-maintainers 

**Changes**
- Add documentation about `req_subkeys`
- Address linter warnings
- Cleanup grammar

**Testing**
- Verified via `make html` and viewing locally in my browser

### CODE REVIEW

**Comment by chunyong-lin** [comment]
LGTM
👏

File: docs/source/rules.rst [hunk 1]
```diff
@@ -13,38 +13,40 @@ Getting Started
...
 
-    @rule('example', 
+    @rule('example',
           logs=[...],
           matchers=[...],
           outputs=[...])
     def example(record):
-        # record analysis             # analyze the incoming record w/ your logic
-        return True                   # return True if an alert should be sent
+        # code                    # analyze the incoming record w/ your logic
```
Comment (strcrzy) [inline docs/source/rules.rst]: should we actually put some sort of example here?

File: docs/source/rules.rst [hunk 2]
```diff
@@ -102,10 +104,31 @@ In the above example, we are evaluating the ``pci`` matcher.  As you can likely
 outputs
 ~~~~~~~
 
-``outputs`` define where the alert should be sent to, if the return value of the function is ``True``.
+``outputs`` define where the alert should be sent to, if the return value of the rule is ``True``.
```
Comment (strcrzy) [inline docs/source/rules.rst]: you capitalize this on L35, but downcase it here. i think you might want to choose either one or the other for consistency

Comment (strcrzy) [inline docs/source/rules.rst]: ("Rule")

**Pull request merged**